### PR TITLE
Update go to 1.15

### PR DESCRIPTION
**Pull request by aledbf**
1. Update k8s dependencies
2. Add additional flags:
- reproducible builds: -trimpath and -ldlflags=-buildid=
- smaller binaries: -w (trim debugger data)

I am opening this PR after receiving a security report for the docker image docker.io/jettech/kube-webhook-certgen v1.3.0                  
https://chartcenter.io/kubernetes-ingress-nginx/ingress-nginx?tab=security

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
 	github.com/imdario/mergo v0.3.7 // indirect
 	github.com/onrik/logrus v0.3.0
 	github.com/sirupsen/logrus v1.4.2
 	github.com/spf13/cobra v0.0.4
-	k8s.io/api v0.18.5
-	k8s.io/apimachinery v0.18.5
-	k8s.io/client-go v0.18.5
+	k8s.io/api v0.19.2
+	k8s.io/apimachinery v0.19.2
+	k8s.io/client-go v0.19.2
```
Comment (vsliouniaev) [inline go.mod:12]: Does this change break someone running k8s <= 1.15? Is there any reason to do it besides just updating dependencies to latest?

Comment (aledbf) [reply go.mod:12]: > Does this change break someone running k8s <= 1.15?

No, there are no breaking changes 

>  Is there any reason to do it besides just updating dependencies to latest?

go 1.15 is supported in k8s v1.19

Comment (vsliouniaev) [reply go.mod:12]: I mean that `admissionregistration/v1beta1` is no longer supported in k8s v1.19, which means this is a breaking change for anyone using a version of Kubernetes < 1.16. Is there a need to upgrade to 0.19.2?

Comment (aledbf) [reply go.mod:12]: That is no correct. The removal has been retargeted to v1.22 https://github.com/kubernetes/kubernetes/issues/82021#issuecomment-636873001

**Pull request merged**